docs/tutorials/Optics/Optics_ani.md

```diff
@@ -0,0 +1,370 @@
+```
+
+We have also made the graph smaller (only goes up to 5eV) in the last line to make it easier to see overlaps (or lack of them). This is the kind of output we get:
+
+![Demonstration of 111](111_comparison.png)
+
+You can try comparing it to any other value, but you'll find it's completely different. So how is this related to the tensor?
+
+Considering the properties of tensors in general, we know that the value from a tensor in any direction is given by the equation
+
+$$ {\epsilon}' = \hat{\mathbf{n}}^T \cdot \mathbf{\epsilon} \hat{\mathbf{n}} $$
+
+Where $\epsilon'$ is the dielectric constant in the direction of $\hat{\mathbf{n}}$, and $\mathbf{\epsilon}$ is the dielectric matrix.
+
+Let's quickly work through what we'd expect in the (111) direction and compare that with what we get from Optados. To get $$ \mathbf{\epsilon} \hat{\mathbf{n}} $$ we do
+
+$$
+\begin{pmatrix}
+\epsilon_{xx} & \epsilon_{xy} & \epsilon_{xz} \\
+\epsilon_{yx} & \epsilon_{yy} & \epsilon_{yz} \\
+\epsilon_{zx} & \epsilon_{zy} & \epsilon_{zz}
+\end{pmatrix}
+\frac{1}{\sqrt3}
+\begin{pmatrix}
+1 \\
+1 \\
+1
+\end{pmatrix}
+= \frac{1}{\sqrt3} \begin{pmatrix}
+\epsilon_{xx} + \epsilon_{xy} + \epsilon_{xz} \\
+\epsilon_{yx} + \epsilon_{yy} + \epsilon_{yz} \\
+\epsilon_{zx} + \epsilon_{zy} + \epsilon_{zz}
+\end{pmatrix}
+$$
+
+Then to get the final result we do
+
+
+
+$$
+\frac{1}{\sqrt3}
+\begin{pmatrix}
+\epsilon_{xx} + \epsilon_{xy} + \epsilon_{xz} \\
+\epsilon_{yx} + \epsilon_{yy} + \epsilon_{yz} \\
+\epsilon_{zx} + \epsilon_{zy} + \epsilon_{zz}
+\end{pmatrix}
+\cdot
+\frac{1}{\sqrt3}
+\begin{pmatrix}
+1 & 1 & 1
+\end{pmatrix}
+= \frac{1}{3} \bigg((\epsilon_{xx} + \epsilon_{xy} + \epsilon_{xz}) + (\epsilon_{yx} + \epsilon_{yy} + \epsilon_{yz}) + (\epsilon_{zx} + \epsilon_{zy} + \epsilon_{zz}) \bigg)
+$$
+
+<a id="final_equation"></a>
+$$
+= \frac{1}{3} \bigg ( \epsilon_{xx} + \epsilon_{yy} + \epsilon_{zz} + 2(\epsilon_{xy} + \epsilon_{yz} + \epsilon_{zx}) \bigg )
+$$
+
+Let's check if this is true using our separated dielectric function tensor files. We will get a function that is equal to the above equation using a Python script, and plot it on the same graph as the case polarised in the (111) direction.
+
+You may use the script
+
+*get_in_direction.py*
+```python
+import numpy as np
+
+# Function to read the data from a file
+def read_data(filename):
+    return np.loadtxt(filename)
+
+eps_xx = read_data("rut_tens1.dat")
+eps_yy = read_data("rut_tens2.dat")
+eps_zz = read_data("rut_tens3.dat")
+eps_xy = read_data("rut_tens4.dat")
+eps_xz = read_data("rut_tens5.dat")
+eps_yz = read_data("rut_tens6.dat")
+
+energies = eps_xx[:, 0]
+
+eps_xx_values = eps_xx[:, 1]
+eps_yy_values = eps_yy[:, 1]
+eps_zz_values = eps_zz[:, 1]
+eps_xy_values = eps_xy[:, 1]
+eps_xz_values = eps_xz[:, 1]
+eps_yz_values = eps_yz[:, 1]
+
+eps_111_values = (eps_xx_values + eps_yy_values + eps_zz_values +
+                  2 * eps_xy_values + 2 * eps_xz_values + 2 * eps_yz_values) / 3
```

Comment:
This could probably be made more readable using dicts.
```suggestion
eps = {key: read_data(f"rut_tens{i}.dat") 
       for i, key in enumerate(('xx', 'yy', 'zz', 'xy', 'xz', 'yz'))}
eps_vals = {key: val[:, 1] for key, val in eps.items()}
energies = eps['xx'][:, 0]

eps_111_values = (sum(eps[ind] for ind in ('xx', 'yy', 'zz')) + 
                  2*sum(eps[ind] for in in ('xy', 'xz', 'yz')) / 3
```